otter/convergence/service.py

```diff
@@ -172,7 +172,8 @@ def update_cache(group, servers, lb_nodes, now):
         sd = thaw(server.json)
         if is_autoscale_active(server, lb_nodes):
             sd["_is_as_active"] = True
-        server_dicts.append(sd)
+        if server.state != ServerState.DELETED:
```

Comment:
I thought this wasn't required since `is_autoscale_active` is already checking for ACTIVE servers